File: app/src/main/java/n7/ad2/storage/ResponsesSourceFactory.java
package n7.ad2.storage;


import android.arch.paging.DataSource;

import n7.ad2.model.ResponseModel;

public class ResponsesSourceFactory extends DataSource.Factory<Integer, ResponseModel> {

    private final ResponsesStorage responsesStorage;
    private String search;

    public ResponsesSourceFactory(ResponsesStorage responsesStorage, String search) {
        this.responsesStorage = responsesStorage;
        this.search = search;
    }

    @Override
    public DataSource<Integer, ResponseModel> create() {
        return new PositionalResponsesDataSource(responsesStorage, search);
    }
}

File: app/src/main/java/n7/ad2/ui/heroResponse/ResponsesSourceFactory.kt
package n7.ad2.ui.heroResponse

import androidx.paging.DataSource
import n7.ad2.ui.heroResponse.domain.vo.VOResponse

class ResponsesSourceFactory(
        private val list: List<VOResponse>,
        private val search: String
) : DataSource.Factory<Int, VOResponse>() {

    override fun create(): DataSource<Int, VOResponse> {
        return PositionalResponsesDataSource(list, search)
    }

}